site/src/components/TemplateLayout/TemplateLayout.tsx

```diff
@@ -50,38 +70,35 @@ export const TemplateLayout: FC<{ children?: JSX.Element }> = ({
 }) => {
   const navigate = useNavigate()
   const styles = useStyles()
-  const organizationId = useOrganizationId()
-  const templateName = useTemplateName()
-  const [templateState, _] = useMachine(templateMachine, {
-    context: {
-      templateName,
-      organizationId,
-    },
-  })
-  const {
-    template,
-    permissions: templatePermissions,
-    getTemplateError,
-  } = templateState.context
-  const permissions = usePermissions()
+  const orgId = useOrganizationId()
+  const { template } = useParams() as { template: string }
+  const templateData = useTemplateData(orgId, template)
+  const dashboard = useDashboard()
 
-  if (getTemplateError) {
+  if (templateData.error) {
     return (
       <div className={styles.error}>
-        <AlertBanner severity="error" error={getTemplateError} />
+        <AlertBanner severity="error" error={templateData.error} />
       </div>
     )
   }
 
-  if (!template || !templatePermissions) {
+  if (templateData.isLoading) {
     return <Loader />
   }
 
+  // Make typescript happy
+  if (!templateData.data) {
+    return <></>
+  }
```

Comment:
Should we just combine this with the `loading` case up above?

Reply:
Yeah, probably better!